app/code/Magento/CatalogUrlRewrite/etc/config.xml

```diff
@@ -9,7 +9,7 @@
     <default>
         <catalog>
             <seo>
-                <generate_category_product_rewrites>1</generate_category_product_rewrites>
+                <generate_category_product_rewrites>0</generate_category_product_rewrites>
```

Comment:
Unfortunately, such changes are backward-incompatible since the default behavior will be changed.